components/net/image_cache_thread.rs

```diff
@@ -258,6 +258,8 @@ struct ImageCache {
     // Images that are loading over network, or decoding.
     pending_loads: AllPendingLoads,
 
+    final_url: Option<ServoUrl>,
```

Comment:
No. That url is the one that was used to initiate the load. The final url is the one that was used to retrieve the final response, after any intermediate redirects.